bitcoinspv/submit_test.go

```diff
@@ -0,0 +1,354 @@
+package bitcoinspv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/gonative-cc/relayer/bitcoinspv/clients/mocks"
+	sui_errors "github.com/gonative-cc/relayer/bitcoinspv/clients/sui"
+	"github.com/gonative-cc/relayer/bitcoinspv/config"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+)
+
+var testSubmitConfig = &config.RelayerConfig{
+	RetrySleepDuration:    1 * time.Millisecond,
+	MaxRetrySleepDuration: 50 * time.Millisecond,
+	HeadersChunkSize:      2,
+	Format:                "auto",
+	Level:                 "debug",
+	NetParams:             "test",
+	BTCCacheSize:          100,
+	BTCConfirmationDepth:  6,
+	ProcessBlockTimeout:   5 * time.Second,
+}
+
+func TestFindFirstNewHeader(t *testing.T) {
+	ctx := context.Background()
+	testBlocks := createTestIndexedBlocks(t, 5, 100) // heights 100, 101, 102, 103, 104
+
+	tests := []struct {
```

Comment:
This should be good if we handle the case findFirstNewHeader returns an error. This can occur in the future when we update our light client package and the version becomes incompatible.

Reply:
Added this case